### PR TITLE
tidy: check fluent files for style

### PR DESCRIPTION
**Pull request by Xiretza**
Inspired by https://github.com/rust-lang/rust/pull/100651#discussion_r947600576

There were a lot of line length violations, so I've excepted that lint - I'm not sure if fluent files can be formatted to avoid long lines at all.

### CODE REVIEW

File: compiler/rustc_error_messages/locales/en-US/borrowck.ftl [hunk 1]
```diff
@@ -13,6 +13,6 @@ borrowck_could_not_normalize =
 
 borrowck_higher_ranked_subtype_error =
     higher-ranked subtype error
-  
+
 generic_does_not_live_long_enough =
```
Comment (davidtwco) [inline compiler/rustc_error_messages/locales/en-US/borrowck.ftl:17]: Could you do a drive-by fix here and change this slug so that it is `borrowck_generic_does_not_live_long_enough`? 

I should have caught this in the pull request that introduced it, we like to start the slugs with the crate name so that they don't overlap between files since it is all one big namespace.

**Review by davidtwco — APPROVED** [review]
This looks good to me, thanks for the patch!

File: compiler/rustc_error_messages/locales/en-US/borrowck.ftl (hunk 1 above)
Comment (Xiretza) [reply compiler/rustc_error_messages/locales/en-US/borrowck.ftl:17]: Done. Seems like I haven't completely understood yet how the translation from slug names to rust items works - the diagnostic used `borrowck::generic_does_not_live_long_enough` all along, so is it just that a leading `${crate_name}_` is chopped off during translation?

Comment (davidtwco) [reply compiler/rustc_error_messages/locales/en-US/borrowck.ftl]: Yeah, we get it from the file name and chop it off. But Fluent only knows the slug here, so when we go ask it to translate a message, if there were two `generic_does_not_live_long_enough` messages in different files, it would get confused.

**Review by davidtwco — APPROVED** [review]
Great! Will r=me once CI passes.

**Comment by davidtwco** [comment]
cc @Mark-Simulacrum for the tidy change

**Comment by est31** [comment]
I'm not sure this PR is a good idea, at least for non-english fluent files, because those will be edited using a tool (at least according to what @Manishearth said), and that tool might not create ftl files that conform with tidy checks. Even with the submodules/subtrees that rustc uses, a lot of them are exempt from tidy, see the `filter_dirs` function in lib.rs, partially because the CI of those projects don't enforce tidy either, so it might cause issues with submodule upgrades. Note that my complaint only applies to the non-english fluent files, as it is my understanding that the english ftl files are kinda special as they will be maintained outside of the tool (or idk).

**Comment by davidtwco** [comment]
> I'm not sure this PR is a good idea, at least for non-english fluent files, because those will be edited using a tool (at least according to what @Manishearth said), and that tool might not create ftl files that conform with tidy checks. Even with the submodules/subtrees that rustc uses, a lot of them are exempt from tidy, see the `filter_dirs` function in lib.rs, partially because the CI of those projects don't enforce tidy either, so it might cause issues with submodule upgrades. Note that my complaint only applies to the non-english fluent files, as it is my understanding that the english ftl files are kinda special as they will be maintained outside of the tool (or idk).

English Fluent will be maintained manually like it is now. We can add whatever exceptions we need for submodules or translations or however we end up handling that, so I think this should be okay.

**Comment by matthiaskrgr** [comment]
@bors r- files are actually causing errors 
https://github.com/rust-lang/rust/pull/100701#issuecomment-1218519364

**Comment by Xiretza** [comment]
My plan is to wait until #100651 lands, then rebase this (fixing the new formatting problems) and hope it doesn't race another PR that adds whitespace problems. Sound sensible?

**Comment by davidtwco** [comment]
I'll approve this but not for inclusion in rollups just in case it gets mixed in with the other translation PRs that are happening and makes things fail.

@bors r+ rollup=never
